firstvoices/backend/search/utils/query_builder_utils.py

```diff
@@ -113,19 +95,61 @@ def get_search_term_query(search_term):
             }
         }
     )
-    return Q(
-        "bool",
-        should=[
+
+    subqueries = [multi_match_query, text_search_field_match_query]
+
+    subquery_domains = {
+        "both": [
             fuzzy_match_title_query,
             exact_match_title_query,
             fuzzy_match_translation_query,
             exact_match_translation_query,
-            multi_match_query,
-            text_search_field_match_query,
         ],
+        "language": [fuzzy_match_title_query, exact_match_title_query],
+        "english": [fuzzy_match_translation_query, exact_match_translation_query],
+    }
+
+    subqueries += subquery_domains.get(domain, [])
+
+    return Q(
+        "bool",
+        should=subqueries,
         minimum_should_match=1,
     )
 
 
 def get_site_filter_query(site_slug):
     return Q("bool", filter=[Q("term", site_slug=site_slug)])
+
+
+# Search params validation
+def get_valid_document_types(input_types):
+    allowed_values = VALID_DOCUMENT_TYPES
+
+    if not input_types:
+        return allowed_values
+
+    values = input_types.split(",")
+    selected_values = [
+        value.strip().lower()
+        for value in values
+        if value.strip().lower() in allowed_values
+    ]
+
+    if len(selected_values) == 0:
+        return None
+
+    return selected_values
+
+
+def get_valid_domain(input_domain_str):
+    string_lower = input_domain_str.lower()
+
+    if (
+        string_lower == "both"
+        or string_lower == "language"
+        or string_lower == "english"
```

Comment:
We use these string values in a couple of important places. Can you please turn them into constants? (Just in this file is fine, I think.)